core/src/main/java/org/opensearch/sql/planner/physical/MLCommonsOperator.java

```diff
@@ -0,0 +1,155 @@
+  private final PhysicalPlan input;
+
+  @Getter
+  private final String algorithm;
+
+  @Getter
+  private final List<Argument> arguments;
+
+  @Getter
+  private final MachineLearningClient machineLearningClient;
+
+  @EqualsAndHashCode.Exclude
+  private Iterator<ExprValue> iterator;
+
+  @Override
+  public void open() {
+    super.open();
+    DataFrame inputDataFrame = generateInputDataset();
+    MLAlgoParams mlAlgoParams = convertArgumentToMLParameter(arguments.get(0), algorithm);
+    MLInput mlinput = MLInput.builder()
+            .algorithm(FunctionName.valueOf(algorithm.toUpperCase()))
+            .parameters(mlAlgoParams)
+            .inputDataset(new DataFrameInputDataset(inputDataFrame))
+            .build();
+    MLPredictionOutput predictionResult = (MLPredictionOutput) machineLearningClient
+            .trainAndPredict(mlinput)
+            .actionGet(30, TimeUnit.SECONDS);
+    Iterator<Row> inputRowIter = inputDataFrame.iterator();
+    Iterator<Row> resultRowIter = predictionResult.getPredictionResult().iterator();
+    iterator = new Iterator<ExprValue>() {
+      @Override
+      public boolean hasNext() {
+          return inputRowIter.hasNext();
+      }
+
+      @Override
+      public ExprValue next() {
+        ImmutableMap.Builder<String, ExprValue> resultBuilder = new ImmutableMap.Builder<>();
+        resultBuilder.putAll(convertRowIntoExprValue(inputDataFrame.columnMetas(),
+                inputRowIter.next()));
+        resultBuilder.putAll(convertRowIntoExprValue(
+                predictionResult.getPredictionResult().columnMetas(),
+                resultRowIter.next()));
+        return ExprTupleValue.fromExprValueMap(resultBuilder.build());
+      }
+    };
+  }
+
+  @Override
+  public <R, C> R accept(PhysicalPlanNodeVisitor<R, C> visitor, C context) {
+    return visitor.visitMLCommons(this, context);
+  }
+
+  @Override
+  public boolean hasNext() {
+    return iterator.hasNext();
+  }
+
+  @Override
+  public ExprValue next() {
+    return iterator.next();
+  }
+
+  @Override
+  public List<PhysicalPlan> getChild() {
+    return Collections.singletonList(input);
+  }
+
+  protected MLAlgoParams convertArgumentToMLParameter(Argument argument, String algorithm) {
+    switch (FunctionName.valueOf(algorithm.toUpperCase())) {
+      case KMEANS:
+        return KMeansParams.builder().centroids((Integer) argument.getValue().getValue()).build();
+
+      default:
+        throw new IllegalArgumentException("unsupported argument type:"
+                + argument.getValue().getType());
+    }
+  }
+
+  private Map<String, ExprValue> convertRowIntoExprValue(ColumnMeta[] columnMetas, Row row) {
+    ImmutableMap.Builder<String, ExprValue> resultBuilder = new ImmutableMap.Builder<>();
+    for (int i = 0; i < columnMetas.length; i++) {
+      ColumnValue columnValue = row.getValue(i);
+      String resultKeyName = columnMetas[i].getName();
+      switch (columnValue.columnType()) {
+        case INTEGER:
```

Comment:
MLCommons supports more data type now, how about adding `short`, `long` and `float` ?

Reply:
Added these three types in next revision